### ISSUE TITLE
Poetry self update --preview installs to system python

### ISSUE THREAD

**Issue by mindlace** [python-poetry/poetry#4728]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: OSX Big Sur 11.6
- **Poetry version**: 1.1.11 & 1.2.0a2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: No pyproject required

## Issue

When updating poetry, it installs to the currently active python rather than poetry's "self contained" venv, leaving the isolated poetry at the prior version.

### Steps to reproduce:

1. `python -m pip list` -> shows no poetry package
2. `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -` -> installs 1.11.1
3. `python -m pip list` -> unchanged
4. `poetry self upgrade --preview` -> upgrades
5. `python -m pip list` -> includes all deps for poetry 1.20.0a2

[Full shell transcript here](
https://gist.github.com/mindlace/1a25012d8d6ef13c96e8f4f98056a79b)

### Expected behavior:

Poetry updates itself inside its virtualenv, and does not install anything in the local/current python env

**Comment by rowanseymour** [comment]
I think we're seeing the same problem with updating from 1.1.11 to 1.1.12. It's trying to update system python and blowing up for a lack of permissions.

```
:~$ which poetry
/home/foo/.local/bin/poetry
:~$ poetry self update
Updating Poetry to 1.1.12

Updating dependencies
Resolving dependencies... (0.7s)

Package operations: 1 install, 19 updates, 0 removals

  - Updating zipp (3.1.0 -> 3.6.0)
  - Updating certifi (2018.1.18 /usr/lib/python3/dist-packages -> 2021.10.8)

  EnvCommandError

  Command ['pip', 'uninstall', 'certifi', '-y'] errored with the following return code 2, and output:
  Found existing installation: certifi 2018.1.18
  Uninstalling certifi-2018.1.18:
  ERROR: Exception:
  Traceback (most recent call last):
    File "/usr/lib/python3.6/shutil.py", line 550, in move
      os.rename(src, real_dst)
  PermissionError: [Errno 13] Permission denied: '/usr/lib/python3/dist-packages/certifi' -> '/tmp/pip-uninstall-viqko_d0'
```

Maybe I'm misunderstanding how it works but I would expect it to only make changes in `/home/foo/.local/lib/python3.6/` and not be touching `/usr/lib/python3/`

**Comment by cpmsmith** [comment]
I'm seeing this as well—somehow, I've ended up with a `poetry` file in `/opt/homebrew/bin` (macOS, arm64) and try as I might, I can't get the uninstaller script to see it.

**Comment by finswimmer** [comment]
Hello,

this is fixed since poetry 1.1.13 by https://github.com/python-poetry/poetry/pull/5048.

fin swimmer
